analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DatabasePasswordsShouldBeSecure.cs

```diff
@@ -0,0 +1,101 @@
+﻿namespace TestFrameworkExtensions
+{
+    using Microsoft.EntityFrameworkCore;
+
+    public class NoncompliantDbContext : DbContext
+    {
+        private DbContextOptionsBuilder builder;
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            optionsBuilder.UseSqlServer("Server=myServerAddress;Database=myDataBase;User Id=myUsername;Password="); // Noncompliant {{Use a secure password when connecting to this database.}}
```

Comment:
Should we consider also [SqlLite](https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.sqlitedbcontextoptionsbuilderextensions.usesqlite?view=efcore-5.0#Microsoft_EntityFrameworkCore_SqliteDbContextOptionsBuilderExtensions_UseSqlite_Microsoft_EntityFrameworkCore_DbContextOptionsBuilder_System_String_System_Action_Microsoft_EntityFrameworkCore_Infrastructure_SqliteDbContextOptionsBuilder__)?

Reply:
we are missing quite a bunch from [the available providers](https://docs.microsoft.com/en-us/ef/core/providers/?tabs=dotnet-core-cli), and yes, it makes sense to cover them

Reply:
I only added SqlLite and OracleClient, there's too many and it would take too much time for the rest, I asked P-L about them anyway

Reply:
Might also be worth to take MongoDb into account. It one of the availble interfaces to use over Azure CosmosDb.